docs/reference/ui/components.md

```diff
@@ -0,0 +1,182 @@
+
+````{note}
+```{include} ../../../client/README.md
+:start-after: <!-- STABILITY:START -->
+:end-before: <!-- STABILITY:END -->
+```
+````
+
+## Fragment
+
+```{eval-rst}
+.. autofunction:: wagtail.admin.templatetags.wagtailadmin_tags.fragment
+```
+
+## Dialog
+
+```{eval-rst}
+.. class:: wagtail.admin.templatetags.wagtailadmin_tags.DialogNode
+   :no-contents-entry:
+```
+
+A dialog to display information in a modal dialog. It is powered by the [`DialogController`](controller:DialogController) (`w-dialog`). To create a dialog, you can use the `{% dialog %}` and `{% enddialog %}` template tags.
+
+```html+django
+{% dialog icon_name="globe" title="Dialog with critical error" id="my-dialog" subtitle="This is a testing subtitle" message_status="critical" message_heading="There was an issue with the thing" message_description="This is a subtext for the message" %}
+    <p>This dialog message was generated by passing message_status=critical as well as message_heading and message_description to the dialog template tag</p>
+{% enddialog %}
+```
+
+Required arguments for the `{% dialog %}` tag:
+
+- `id`: The ID of the dialog, to be used by the [`{% dialog_toggle %}`](dialog_toggle) tag.
+- `title`: The title of the dialog.
+
+Optional arguments for the `{% dialog %}` tag:
+
+- `dialog_root_selector`: The CSS selector for the dialog root element, defaults to `body`. Useful when you want to render the dialog in a specific part of the DOM, such as within a `<form>` element.
+- `icon_name`: The name of the icon to display in the dialog header.
+- `subtitle`: The subtitle of the dialog.
+- `theme`: The theme of the dialog, either empty or `floating`.
+- `classname`: CSS classes for styling the dialog.
+
+The content of the dialog can be any Django template code written between the `{% dialog %}` and `{% enddialog %}` tags.
+
+The dialog can also display a status message at the top, such as an error or a warning. The following optional arguments can be used to render the message:
+
+- `message_status`: The status level of the message, which can be one of: `info`, `warning`, `critical`, `success`.
+- `message_heading`: The heading of the message.
+- `message_description`: The description of the message.
+
+### Dialog toggle
+
+```{eval-rst}
+.. function:: wagtail.admin.templatetags.wagtailadmin_tags.dialog_toggle
+   :no-contents-entry:
+```
+
+A button component that opens a dialog with the specified ID. To create a button that opens a dialog, you can use the `{% dialog_toggle %}` template tag.
+
+```html+django
+{% dialog_toggle classname='button' dialog_id='my-dialog' text='Dialog with error' %}
+```
+
+Arguments for the `{% dialog_toggle %}` tag:
+
+- `dialog_id`: The ID of the dialog to open.
+- `text`: The text to display on the button.
+- `classname` (optional): CSS classes for styling the button.
+
+## Dropdown
```

Comment:
For this (and "Button with dropdown"), I'm not sure whether we should document the template tags or the Python component versions, or both?

We have them in https://github.com/wagtail/wagtail/blob/main/wagtail/admin/widgets/button.py, which would also allow us to sidestep the issue with having to document the CSS classes. However, that module still has some pending deprecations, which aren't a blocker to us documenting the non-deprecated classes, but I just thought I'd mention. Also, ideally that module shouldn't live in `wagtail.admin.widgets` because the other modules in there are for form widgets, and buttons are not that. It should probably live in `wagtail.admin.ui` instead.